src/libs/conduit/conduit_data_array.cpp

```diff
@@ -1436,6 +1511,135 @@ DataArray<T>::compact_elements_to(uint8 *data) const
 }
 
 
+//---------------------------------------------------------------------------//
+template <typename T> 
+std::string
+DataArray<T>::to_summary_string_default() const
+{ 
+    return to_summary_string();
+}
+
+//---------------------------------------------------------------------------//
+template <typename T> 
+std::string
+DataArray<T>::to_summary_string(index_t threshold) const
+{ 
+    std::ostringstream oss;
+    to_summary_string_stream(oss, threshold);
+    return oss.str();
+}
+
+//---------------------------------------------------------------------------//
+template <typename T> 
+void
+DataArray<T>::to_summary_string_stream(std::ostream &os,
+                                       index_t threshold) const
+{ 
+    // if we are less than or equal to threshold, we use to_yaml
+    index_t nele = number_of_elements();
+
+    if(nele <= threshold)
+    {
+        to_yaml_stream(os);
+    }
+    else
+    {
+        // if above threshold only show threshold # of values
+        int half = threshold / 2;
+        int bottom = half;
+        int top = half;
+
+        //
+        // if odd, show 1/2 +1 first
+        //
+
+        if( (threshold % 2) > 0)
+        {
+            bottom++;
+        }
+
+        if(nele > 1)
+            os << "[";
+
+        bool first = true;
```

Comment:
To reduce the number of variables that we need to keep track of, would it make sense to use `idx == 0` as the check for the first element instead of a separate `first` variable?

Reply:
agreed, done